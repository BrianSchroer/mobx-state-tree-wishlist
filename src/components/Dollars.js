import React from 'react';

const Dollars = ({ amount }) => amount.toLocaleString('en-US', { style: 'currency', currency: 'USD' });

export default Dollars;