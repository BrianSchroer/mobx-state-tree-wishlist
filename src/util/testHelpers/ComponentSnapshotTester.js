import React from 'react';
import renderer from 'react-test-renderer';

/**
 * React component snapshot helper
 */
export default class ComponentSnapshotTester {

  /**
   * Create new ComponentSnapshotTester instance
   * @param {*} component - React component
   */
  constructor(component) {
    this.component = component.type;
    this.props = component.props;
    this.adjustProps = (props, propOverrides) => Object.assign({}, props, propOverrides);
  }

  /**
   * Specify callback function to be called by the test function to adjust component
   * props with propOverrides.
   * (Defaults to adjusting top-level props.)
   */
  withPropAdjustor = (adjustor) => {
    this.adjustProps = adjustor;
    return this;
  };

  /**
   * Adjust props that were passed to the constructor with propOverrides, then create
   * snapshot and assert that it matches previously accepted screenshot.
   */
  test = (propOverrides) => {
    const Component = this.component;
    const props = this.adjustProps(this.props, propOverrides);

    const newSnapshot = renderer.create(<Component {...props} />).toJSON();

    expect(newSnapshot).toMatchSnapshot();
  }

  /**
   * Create snapshot and assert that it matches previously accepted screenshot. 
   * @param {*} component - React component1 
   */
  static test(component) {
    new ComponentSnapshotTester(component).test();
  }
}