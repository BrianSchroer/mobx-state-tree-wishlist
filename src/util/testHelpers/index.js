export { default as mobxSnapshotHelper } from './mobxSnapshotHelper';
export { default as reactRendererSnapshotHelper } from './reactRendererSnapshotHelper';