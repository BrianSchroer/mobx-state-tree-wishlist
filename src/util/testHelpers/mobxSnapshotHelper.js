import { getSnapshot } from 'mobx-state-tree';

export default class {
  static createSnapshot(target) {
    return getSnapshot(target);
  }

  static assertMatch(target) {
    expect(this.createSnapshot(target)).toMatchSnapshot();
  }
}