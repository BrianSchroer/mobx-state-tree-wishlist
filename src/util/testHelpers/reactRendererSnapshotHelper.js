import renderer from 'react-test-renderer';

/**
 * react-test-renderer snapshot helper methods
 */
export default class {

  /**
   * Create component from JSX
   * @param {JSX} componentJsx (e.g. '<Button />')
   * @returns {object} component object
   */
  static createComponent(componentJsx) {
    return renderer.create(componentJsx);
  }

  /**
   * Create component from JSX and serialize as JSON "snapshot" string.
   * @param {JSX} componentJsx (e.g. '<Button />')
   * @returns {string} JSON string.
   */
  static createSnapshot(componentJsx) {
    return this.createComponent(componentJsx).toJSON();
  }

  /**
   * Create component from JSX, serialize as JSON string, and compare to accepted snapshot.
   * If a snapshot doesn't already exist, one is saved. If a previous snapshot exists
   * and the new snapshot doesn't match, you'll be asked if that's an error or if you want
   * to accept the new snapshot.
   * @param {JSX} componentJsx (e.g. '<Button />')
   */
  static assertMatch(componentJsx) {
    expect(this.createSnapshot(componentJsx)).toMatchSnapshot();
  }
}