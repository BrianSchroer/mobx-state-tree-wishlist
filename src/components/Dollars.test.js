import React from 'react';
import Dollars from './Dollars';
import { reactRendererSnapshotHelper as snapshotHelper } from '../util/testHelpers';

describe('Dollars', () => {
  [0, 1, 1.23, 12.34, 123.45, 1234.56].forEach(amount => {
    it(`should render ${amount} correctly`, () => {
      snapshotHelper.assertMatch(<Dollars amount={amount} />);
    });
  })
});