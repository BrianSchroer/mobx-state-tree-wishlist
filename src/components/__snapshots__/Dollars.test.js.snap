// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dollars should render 0 correctly 1`] = `"$0.00"`;

exports[`Dollars should render 1 correctly 1`] = `"$1.00"`;

exports[`Dollars should render 1.23 correctly 1`] = `"$1.23"`;

exports[`Dollars should render 12.34 correctly 1`] = `"$12.34"`;

exports[`Dollars should render 123.45 correctly 1`] = `"$123.45"`;

exports[`Dollars should render 1234.56 correctly 1`] = `"$1,234.56"`;
